ethereum/dataproviders/src/main/java/tech/pegasys/teku/dataproviders/lookup/BlockProvider.java

```diff
@@ -39,21 +38,18 @@ static BlockProvider fromDynamicMap(Supplier<Map<Bytes32, SignedBeaconBlock>> ma
   }
 
   static BlockProvider fromMapWithLock(
-      final Map<Bytes32, SignedBeaconBlock> blockMap,
-      final AsyncRunner asyncRunner,
-      final Lock readLock) {
-    return (roots) ->
-        asyncRunner.runAsync(
-            () -> {
-              readLock.lock();
-              try {
-                return roots.stream()
-                    .filter(blockMap::containsKey)
-                    .collect(Collectors.toMap(Function.identity(), blockMap::get));
-              } finally {
-                readLock.unlock();
-              }
-            });
+      final Map<Bytes32, SignedBeaconBlock> blockMap, final Lock readLock) {
+    return (roots) -> {
+      readLock.lock();
+      try {
+        return SafeFuture.completedFuture(
+            roots.stream()
+                .filter(blockMap::containsKey)
+                .collect(Collectors.toMap(Function.identity(), blockMap::get)));
```

Comment:
worth re-using fromMap inside here?

Reply:
I don't understand how it's possible?

Reply:
this is redundant if we do the other comment...